bee-storage/bee-storage/src/access/batch.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::storage::Backend;
+
+#[async_trait::async_trait]
+/// BatchBuilder trait will extend the Backend with Batch builder functionality,
+/// therefore it should be explicitly implemented for the corresponding Backend.
+pub trait BatchBuilder: Backend + Sized {
+    /// Batch type acts like memory buffer which queue all the write operations.
+    type Batch: Default + Send + Sized;
+
+    /// This method will create and return the constraint Batch object
+    fn batch_begin() -> Self::Batch {
```

Comment:
Any reason why this isn't `begin_batch`? Seems like a weird ordering. Ditto `batch_commit`.

Reply:
@thibault-martinez please resolve this.

Reply:
what about having a `begin_batch`, `commit_batch`, `insert_to_batch` and a `delete_from_batch`?

Reply:
Depending on whether `batch` would be used as the *object* or as part of the *verb*, I'ld go with Josh calling the function `begin_batch` (batch is the object), or `batch_begin` (batch is part of the verb). In the latter case something else is begun, that might be omitted because it's obvious from the context. That is something @LouayKamel has to decide.

Reply:
But from the docs it seems like the object is the `batch` itself, and hence `begin_batch`, `create_batch` or `new_batch` would be cleaner.